lib/phase/flatten.js

```diff
@@ -2,9 +2,32 @@
 
 const Hoek = require('hoek');
 const clone = require('clone');
+const TEMPLATE_REGEX = require('screwdriver-data-schema').config.regex.FULL_TEMPLATE_NAME;
+const NAME_MATCH = 1;
+const VERSION_MATCH = 3;
+const LABEL_MATCH = 7;
 
 /**
- * Overlays the stuff specified in a specific job on top of the defaults in shared
+ * Merge oldJob into newJob
+ * "oldJob" takes precedence over "newJob". For ex: individual job settings vs shared settings
+ * @param  {Object}   newJob    Job to be merged into. For ex: shared settings
+ * @param  {Object}   oldJob    Job to merge. For ex: individual job settings
+ */
+function merge(newJob, oldJob) {
+    // Intialize new job with default fields
+    newJob.environment = newJob.environment || {};
+    newJob.secrets = newJob.secrets || [];
+    newJob.settings = newJob.settings || {};
+
+    // Merge
+    Object.assign(newJob.environment, oldJob.environment || {});
+    Object.assign(newJob.settings, oldJob.settings || {});
+    newJob.secrets = newJob.secrets.concat(oldJob.secrets || []);
```

Comment:
Yeah we should remove duplicates. Currently, it doesn't when we merge `shared` and individual job:
<img width="576" alt="screen shot 2017-03-14 at 2 23 30 pm" src="https://cloud.githubusercontent.com/assets/3401924/23922967/df920878-08c1-11e7-9e12-11628d489ddd.png">

I should fix that.